### PR TITLE
Modify guest installation schedule for guest migration

### PR DESCRIPTION
**Pull request by RoyCai7**
poo#https://progress.opensuse.org/issues/124928
Modify guest installation schedule for guest migration to keep job running well.

- Related ticket: https://progress.opensuse.org/issues/124928
- Needles: N/A
- Verification run: [kvm test](http://openqa.qam.suse.cz/tests/52715), [xen test](http://openqa.qam.suse.cz/tests/52713), [kvm migration test](http://openqa.qam.suse.cz/tests/overview?build=:S:M:28650:295434:64bit-guest-mi-ipmi:12-SP5:g_m_test:&distri=sle&version=12-SP5&groupid=149),  [xen migration test](http://openqa.qam.suse.cz/tests/overview?build=:S:M:28650:295434:64bit-guest-mi-ipmi-group2:12-SP5:g_m_test:&distri=sle&version=12-SP5&groupid=160)

### CODE REVIEW

**Comment by waynechen55** [comment]
Could you help modify existing one ```schedule/qam/common/virt/virt_guest_migration.yaml``` instead of adding new schedule file ?

**Comment by RoyCai7** [comment]
> Could you help modify existing one `schedule/qam/common/virt/virt_guest_migration.yaml` instead of adding new schedule file ? This one is a fake one to prevent CI checking failure, it can still be employed.

But they are two completely different files.
This is virt_guest_migration.yaml file:
```
---
name: virt_guest_migration.yaml
description: |
  Maintainer: Wayne Chen (wchen@suse.com) qe-virt@suse.de
  Yaml scheduling file for guest migration test
schedule:
  - "{{create_barriers}}"
  - virt_autotest/login_console
  - "{{guest_migration}}"
conditional_schedule:
  create_barriers:
    VIRT_NEW_GUEST_MIGRATION_SOURCE:
      0:
        - virt_autotest/parallel_guest_migration_barrier
  guest_migration:
    VIRT_NEW_GUEST_MIGRATION_SOURCE:
      1:
        - virt_autotest/parallel_guest_migration_source
      0:
        - virt_autotest/parallel_guest_migration_destination
```

Current file qam_virt_migration.yaml is a little same schedule/qam/common/virt/with qam_virt_install_guest. 
Tendor and I made some changes before.
refer poo#https://progress.opensuse.org/issues/124928

**Comment by waynechen55** [comment]
> > Could you help modify existing one `schedule/qam/common/virt/virt_guest_migration.yaml` instead of adding new schedule file ? This one is a fake one to prevent CI checking failure, it can still be employed.
> 
> But they are two completely different files. This is virt_guest_migration.yaml file:
> 
> ```
> ---
> name: virt_guest_migration.yaml
> description: |
>   Maintainer: Wayne Chen (wchen@suse.com) qe-virt@suse.de
>   Yaml scheduling file for guest migration test
> schedule:
>   - "{{create_barriers}}"
>   - virt_autotest/login_console
>   - "{{guest_migration}}"
> conditional_schedule:
>   create_barriers:
>     VIRT_NEW_GUEST_MIGRATION_SOURCE:
>       0:
>         - virt_autotest/parallel_guest_migration_barrier
>   guest_migration:
>     VIRT_NEW_GUEST_MIGRATION_SOURCE:
>       1:
>         - virt_autotest/parallel_guest_migration_source
>       0:
>         - virt_autotest/parallel_guest_migration_destination
> ```
> 
> Current file qam_virt_migration.yaml is a little same with schedule/qam/common/virt/with qam_virt_install_guest. Tendor and I made some changes before. refer poo#https://progress.opensuse.org/issues/124928

I do not think ```schedule/qam/common/virt/virt_guest_migration.yaml``` is needed anymore, you can rename and modify it at your will I think.

**Comment by waynechen55** [comment]
S

> > Could you help modify existing one `schedule/qam/common/virt/virt_guest_migration.yaml` instead of adding new schedule file ? This one is a fake one to prevent CI checking failure, it can still be employed.
> 
> But they are two completely different files. This is virt_guest_migration.yaml file:
> 
> ```
> ---
> name: virt_guest_migration.yaml
> description: |
>   Maintainer: Wayne Chen (wchen@suse.com) qe-virt@suse.de
>   Yaml scheduling file for guest migration test
> schedule:
>   - "{{create_barriers}}"
>   - virt_autotest/login_console
>   - "{{guest_migration}}"
> conditional_schedule:
>   create_barriers:
>     VIRT_NEW_GUEST_MIGRATION_SOURCE:
>       0:
>         - virt_autotest/parallel_guest_migration_barrier
>   guest_migration:
>     VIRT_NEW_GUEST_MIGRATION_SOURCE:
>       1:
>         - virt_autotest/parallel_guest_migration_source
>       0:
>         - virt_autotest/parallel_guest_migration_destination
> ```
> 
> Current file qam_virt_migration.yaml is a little same with schedule/qam/common/virt/with qam_virt_install_guest. Tendor and I made some changes before. refer poo#https://progress.opensuse.org/issues/124928

Sorry, ```schedule/qam/common/virt/virt_guest_migration.yaml``` is a different file. But you can still make change to it, for example, move it to a more suitable location or move your new yaml file to the same location as this one at your will.

**Comment by RoyCai7** [comment]
@tbaev @tonyyuan1 Please also help review.

**Comment by waynechen55** [comment]
> which place is better, do you think? @waynechen55 the put the both file. I also think it is ok to put them in the folder: chedule/qam/common/virt/.

That is OK if they are all in ```schedule/qam/common/virt```.

**Comment by tbaev** [comment]
I too think it is best to keep the two separate files in `schedule/qam/common/virt` 

Merging it to one file, is not so flexible if tests change down the line

**Comment by tbaev** [comment]
`schedule/qam/common/virt/` is a good place for the schedule file.

I am okay with the name `qam_virt_guest_migration.yaml` is the yaml used for guest installation on host, and the name `virt_guest_migration.yaml` is the yaml used for guest migration test suit. I think it is better to keep the two files separated.

It is possible to combine   `qam_virt_guest_migration.yaml` and `qam_virt_install_guest.yaml` into one schedule file, but I think it is better to keep them separated too.

File: schedule/qam/common/virt/qam_virt_migration.yaml
Comment (alice-suse) [inline schedule/qam/common/virt/qam_virt_migration.yaml:3]: Please also add maintainer info here -- your name <mail>, qe-virt@suse.de

Comment (alice-suse) [reply schedule/qam/common/virt/qam_virt_migration.yaml]: Besides, based on the description and the contents of the file, indeed this file manages the schedule for mainly the steps after host installation and before migration, how about making the name of the file `qam_virt_migration.yaml` more reflecting what it does?

**Comment by alice-suse** [comment]
> It is possible to combine `qam_virt_guest_migration.yaml` and `qam_virt_install_guest.yaml` into one schedule file, but I think it is better to keep them separated too.

It is usually good practice to think one step further when making code decisions :). Maybe here you can consider too what's longer term better solution to save maintain effort.

**Comment by RoyCai7** [comment]
I will try combine them into one schedule file. 
Need more verification runs. It looks not complex.

**Comment by RoyCai7** [comment]
After combining, need add INSTALL_GUEST=1 in the testsuit (qam-xen-install, qam-kvm-install, qam_virt_xen_install_guests_migration_src, qam_virt_kvm_install_guests_migration_src)
add INSTALL_GUEST=0 in the testsuit(qam_virt_xen_install_guests_migration_dst, qam_virt_kvm_install_guests_migration_dst)

In the meanwhile, delete parameter: VIRT_NEW_GUEST_MIGRATION_SOURCE in the testsuit.

**Comment by tbaev** [comment]
Looks good to me, after merge INSTALL_GUEST=1 has to be added to testsuit AND changing the test suit from `YAML_SCHEDULE=schedule/qam/common/virt/qam_virt_migration.yaml` to this one.

**Comment by RoyCai7** [comment]
> `ML_SCHEDULE=schedule/qam/common/virt/qam_virt_migration.yaml` to this one.

Done and passed VR tests.

**Review by alice-suse — APPROVED** [review]
LGTM.

If no further comments, will merge it EOB today.

**Pull request merged**
